README.md

```diff
@@ -50,6 +52,130 @@ import {
+
+```js
+const { hash, sections } = useContext(AnchorContext);
+```
+
+Here is its typing :
+
+| Key               |                      Type                       |                                                                                                                      Description |
+| ----------------- | :---------------------------------------------: | -------------------------------------------------------------------------------------------------------------------------------: |
+| sections          |                 `HTMLElement[]`                 |                                      List of the registered sections elements that we watch, in our codebase it is AnchorSection |
+| hash              |                    `string`                     |                                                                 The registered hash corresponding to our current scroll position |
+| registerSection   |        `(element: HTMLElement) => void`         | Function to add a Section to our sections list, our scrollEvent listener will then update the hash if the section is scrolled to |
+| unregisterSection |        `(element: HTMLElement) => void`         |                 Function to remove a Section to our sections list, our scrollEvent listener will then stop checking this section |
+| setHash           | `(hash: string, withScroll?: boolean) => void;` |                              Setter function from the internal useHash hooks, use it to programmatically change the current hash |
+
+#### AnchorSection
+
+AnchorSection is our most used components, it defines the scroll position you will arrive to on navigation/reloading
+
+```js
+<AnchorSection className="dBlock anchor" id="definitions" />
+```
+
+Its props are the usual HTMLElement's props (`className, data-*`), along with an id used for recognizing the update the current hash on scroll.
+
+```js
+interface TProps extends React.HTMLAttributes<HTMLElement> {
+  id: string;
+}
+```
+
+Internally it creates a `<b/>` tag to which we scroll to on reload and detect if we scrolled past it.
+
+```JSX
+<>
+  <b {...attributes}></b>
+  {...children}
+</>
+```
+
+#### AnchorLink
+
+AnchorLink is a component made to have an activeClassname if its `href` is the current hash
+
+```js
+interface TProps extends React.AnchorHTMLAttributes<HTMLAnchorElement> {
+  children: React.ReactNode[] | React.ReactNode;
+  activeClassName?: string;
+}
+```
+
+```JSX
+<AnchorLink className="dTable w100 pad15" href="#definitions" activeClassName="blue">
+```
+
+### Advanded Usage
+
+Here will be informations about our internals processing hooks. For most use and without needs for great customizations, you will not need to import those hooks.
+
+#### useHash
+
+useHash is a custom react hooks used internally in AnchorProvider. We still export it if you want to have an advanced usage (Otherwise it should not be needed) and not use AnchorProvider.
```

Comment:
```suggestion
useHash is a custom React hook used internally in AnchorProvider. We still export it if you want to have an advanced usage (Otherwise it should not be needed) and not use AnchorProvider.
```